src/index.ts

```diff
@@ -591,22 +471,86 @@ class LayercodeClient implements ILayercodeClient {
     return this.wavRecorder.getStream();
   }
 
+  async setInputDevice(deviceId: string): Promise<void> {
+    try {
+      this.deviceId = deviceId;
+
+      // Restart recording with the new device
+      await this._restartAudioRecording();
+
+      // Reinitialize VAD with the new audio stream if VAD is enabled
+      if (this.vadManager.isVADEnabled()) {
+        console.log('Reinitializing VAD with new audio stream');
+        const newStream = this.wavRecorder.getStream();
+        this.vadManager.reinitialize(newStream);
+      }
+
+      // Set up amplitude monitoring
+      this._setupAmplitudeMonitoring(this.wavRecorder, this.options.onUserAmplitudeChange, (amp) => (this.userAudioAmplitude = amp));
+
+      console.log(`Successfully switched to input device: ${deviceId}`);
+    } catch (error) {
+      console.error(`Failed to switch to input device ${deviceId}:`, error);
+      throw new Error(`Failed to switch to input device: ${error instanceof Error ? error.message : String(error)}`);
+    }
+  }
+
   /**
-   * Switches the input device for the microphone and restarts recording
-   * @param {string} deviceId - The deviceId of the new microphone
+   * Manually switches to the next available audio input device
    */
-  async setInputDevice(deviceId: string): Promise<void> {
-    if (this.wavRecorder) {
-      try {
-        await this.wavRecorder.end();
-      } catch (e) {}
+  private async _switchToNextDevice(): Promise<void> {
+    try {
+      const devices = await this.wavRecorder.listDevices();
+      const audioInputs = devices.filter((d) => d.kind === 'audioinput');
+      const nextDevice = audioInputs.find((d) => d.deviceId !== this.deviceId);
```

Comment:
Should we just choose the new default device, instead of the first device in the list (that isn't the old one)